src/hdusd/utils/matlib.py

```diff
@@ -143,29 +187,53 @@ def __init__(self, mat_json):
         for id in mat_json['packages']:
             self.packages.append(Package(id))
 
+    def get_description(self):
+        description = f"{self.title}"
+        if self.description:
+            description += f"\n{self.description}"
+        if self.category:
+            description += f"\nCategory: {self.category.title}"
+        description += f"\nAuthor: {self.author}"
+
+        return description
+
     @classmethod
     def get_materials(cls, limit=10, offset=0):
+        log.debug("Requesting material",
+            f"url: {URL}/materials, params: 'limit': {limit}, 'offset': {offset}")
         response = requests.get(f"{URL}/materials", params={'limit': limit, 'offset': offset})
         res_json = response.json()
+        log.debug("Request done")
+
         for mat_json in res_json['results']:
             mat = Material(mat_json)
             if not mat.packages:
                 continue
-
             yield mat
 
     @classmethod
     def get_all_materials(cls):
         offset = 0
-        limit = 10
+        limit = 500
 
-        while True:
-            mat = None
-            for mat in cls.get_materials(limit, offset):
-                yield mat
+        log.debug("Requesting material",
+            f"url: {URL}/materials, params: 'limit': {limit}, 'offset': {offset}")
+        response = requests.get(f"{URL}/materials", params={'limit': limit, 'offset': offset})
+        res_json = response.json()
+        log.debug("Request done")
```

Comment:
think of moving such requesting procedure to move into separate function, probably request_json()